File: Task 1/C++/Components/Product/Consumable.cpp
#include "Consumable.h"

Consumable::Consumable () : Product(){
    calories = 0;
}

Consumable::Consumable (std::string name, float cost, std::string supplier, float discount_amount, int packing_units, float cost_per_unit, int calories)
            : Product(name, cost, supplier, discount_amount, packing_units, cost_per_unit){
    this->calories = calories;
}

Consumable::Consumable(const Consumable& other) : Product(other) {
    this->calories = other.calories;
}

std::string Consumable::display_product_info(){
    return Product::display_product_info() + 
            "\nCalories: " + std::to_string(calories);
}

File: Task 1/Java/logistics/src/main/java/com/logistics/Components/Product/Consumable.java
package com.logistics.Components.Product;

public class Consumable extends Product {

    protected int calories;

    public Consumable() {
        super();
        calories = 0;
    }

    public Consumable(String name, float cost, String supplier, float discountAmount, int packingUnits, float costPerUnit, int calories) {
        super(name, cost, supplier, discountAmount, packingUnits, costPerUnit);
        this.calories = calories;
    }

    // Copy Constructor
    public Consumable(Consumable other) {
        super(other);
        this.calories = other.calories;
    }

    // Clone method
    @Override
    public Consumable clone() {
        return new Consumable(this);
    }

    // Getters
    public int getCalories() { return calories; }

    // Setters
    public void setCalories(int calories) { this.calories = calories; }

    @Override
    public String display_product_info() {
        return super.display_product_info() + "\nCalories: " + calories;
    }
}
